### PR TITLE
Feat/xpub UI

### LINKED ISSUE #3454: Confusing UI for Account XPUB
When the device is unplugged, the `Show public key` button looks like this:

This is very confusing UI not following the least surprise principle.

Reasons:

* when the device is unplugged the button shows the spinner - this makes me think I have to wait, not to plug in the device
* when the device is plugged in - the XPUB is shown without the confirmation on the device

I think this workflow should follow the Show Address UI:
* allow showing the XPUB without the device connected, but with the warning
* when the device is connected, show the XPUB, but also show it on the device to confirm it

### PR DESCRIPTION
**Pull request by komret**
## Description

- [x] enable displaying XPUB while device is not connected
- [x] display XPUB on device for confirmation when device is connected
- [x] show warning before displaying in device is not backed up (@mroz22, please check the changes in connect)
- [x] enable closing the no-backup modal for both XPUB and receive address flow and add anchor to the backup link
- [x] unit tests

## Related Issue

Resolve [#3454](https://github.com/trezor/trezor-suite/issues/3454)

#### Device not backed up:
Screenshot 2023-05-31 at 13 22 22
#### Confirmation on device:
Screenshot 2023-05-31 at 15 37 04
#### Device not connected:
Screenshot 2023-05-31 at 15 35 43
Screenshot 2023-05-31 at 15 33 55

### CODE REVIEW

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                                variant="tertiary"
+                                onClick={copy}
+                                data-test={copyButtonDataTest}
+                            >
+                                {copyButtonText}
+                            </Button>
+                        )}
+                    </CopyButtonWrapper>
+                ) : (
+                    <StyledDeviceDisconnected label={device.label} />
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: `border-radius: 8px;`

Noticed that it's not there on the design but it looks off

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            onCancel={onCancel}
+            description={
+                <Translation
+                    id={warningText}
+                    values={{ claim: <Translation id={description} /> }}
+                />
+            }
+            bottomBar={
+                <>
+                    <Button variant="secondary" onClick={showUnverifiedAndContinue}>
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:82]: nit: `flex: 1` looks nicer

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import { useDevice, useDispatch } from '@suite-hooks';
+import { ThunkAction } from '@suite-types';
+import { Button, Image, ModalProps } from '@trezor/components';
+
+const ImageWrapper = styled.div`
+    padding: 60px 0px;
+`;
+
+const StyledModal = styled(Modal)`
+    width: 600px;
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx]: nit: `500px` looks nicer

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import { applySettings } from '@settings-actions/deviceSettingsActions';
+import { Translation, Modal } from '@suite-components';
+import { TranslationKey } from '@suite-components/Translation';
+import { useDevice, useDispatch } from '@suite-hooks';
+import { ThunkAction } from '@suite-types';
+import { Button, Image, ModalProps } from '@trezor/components';
+
+const ImageWrapper = styled.div`
+    padding: 60px 0px;
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx]: nit: `20px 0px` looks nicer

image

File: packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts [hunk 5]
```diff
@@ -73,7 +73,7 @@ export const verifyAddress =
 
         const modalPayload = {
             device,
-            address,
+            value: address,
```
Comment (dahaca) [inline packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts:76]: Why this change?

File: packages/suite/src/actions/wallet/publicKeyActions.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import * as modalActions from '@suite-actions/modalActions';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import { GetState, Dispatch } from '@suite-types';
+import TrezorConnect from '@trezor/connect';
+
+export const openXpubModal =
+    (params?: Partial<Extract<modalActions.UserContextPayload, { type: 'xpub' }>>) =>
```
Comment (dahaca) [inline packages/suite/src/actions/wallet/publicKeyActions.ts]: narrow down the type to `isConfirmed` and  `isCancelable` only maybe?

File: packages/suite/src/actions/wallet/publicKeyActions.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    // Prevent flickering screen when modal changes.
+    dispatch(modalActions.preserve());
+
+    const params = {
+        device,
+        path: account.path,
+        useEmptyPassphrase: device.useEmptyPassphrase,
+        showOnTrezor: true,
+    };
+    let response;
```
Comment (dahaca) [inline packages/suite/src/actions/wallet/publicKeyActions.ts]: nit: type response as `Success<unknown> | Unsuccessful`?

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            }
+            isCancelable={isCancelable}
+            onCancel={onCancel}
+        >
+            <Wrapper>
+                <QrCode value={value} />
+                <Value data-test={valueDataTest}>{value}</Value>
+                {device.connected ? (
+                    <CopyButtonWrapper ref={htmlElement}>
+                        {isConfirmed && (
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: makes little sense to not show the copy button if the device is not connected IMO

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    extends Pick<ModalProps, 'heading' | 'isCancelable' | 'onCancel'> {
+    device: TrezorDevice;
+    copyButtonText: React.ReactNode;
+    copyButtonDataTest?: string;
+    isConfirmed?: boolean;
+    value: string;
+    valueDataTest?: string;
+}
+
+export const ConfirmDeviceScreen = ({
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: Sound a bit like `DevicePromptModal` to me, quite generic. What about `ConfirmValueScreen` or `ConfirmValueOnDevice`?

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    isConfirmed,
+    onCancel,
+    value,
+    valueDataTest,
+}: ConfirmDeviceScreenProps) => {
+    const htmlElement = useRef<HTMLDivElement>(null);
+
+    const dispatch = useDispatch();
+
+    const copy = () => {
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: `document.execCommand` is deprecated, so what about `navigator.clipboard.writeText(text)`? Could be added directly to `copyToClipboard` maybe? @matejkriz  Eliminates the need for a `ref` too.

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    showUnverifiedButtonText,
+    onCancel,
+    showUnverified,
+    verify,
+    warningText,
+}: ConfirmUnverifiedProps) => {
+    const dispatch = useDispatch();
+    const { device, isLocked } = useDevice();
+
+    // just to make TS happy
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:41]: we all want it to be happy ^^

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+                        variant="primary"
+                        onClick={verifyAndContinue}
+                        isDisabled={isDeviceLocked}
+                    >
+                        <Translation id={primaryButtonText} />
+                    </Button>
+                </>
+            }
+        >
+            <ImageWrapper>
```
Comment (dahaca) [inline packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx]: nit: add margins to the image instead?

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx (hunk 8 above)
Comment (dahaca) [reply packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: Consulted ondra and he agrees that the copy button can be rendered right away in case the device is not connected.

Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: Agreed, done.

File: packages/suite/src/actions/wallet/publicKeyActions.ts (hunk 6 above)
Comment (komret) [reply packages/suite/src/actions/wallet/publicKeyActions.ts]: Done.

File: packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts (hunk 5 above)
Comment (komret) [reply packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts:76]: It allowed me to spread the payload in [UserContextModal](https://github.com/trezor/trezor-suite/pull/8613/files#diff-ec45a7963475465fb76fbb83c0c32f3368eb57efb9b74ca644f0dfb6f50efcbf), wasn't really necessary...

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx (hunk 2 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:82]: Agreed, but giving it to the primary button instead.

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx (hunk 3 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx]: Done.

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx (hunk 4 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx]: Done.

File: packages/suite/src/actions/wallet/publicKeyActions.ts (hunk 7 above)
Comment (komret) [reply packages/suite/src/actions/wallet/publicKeyActions.ts]: Done.

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx (hunk 9 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: Renamed to `ConfirmValueOnDevice`.

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx (hunk 11 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:41]: I left that comment from a previous version :)

File: packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx (hunk 12 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx]: Done.

File: packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx (hunk 10 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: You're right, refactored!

Comment (dahaca) [reply packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx]: Thank you for spending time on it!

**Comment by komret** [comment]
Added an option to close the modal while confirming on device.

Also moved the warning to the top, it looks better now that there is the copy button:
Screenshot 2023-06-05 at 19 17 48

**Review by dahaca — APPROVED** [review]
Thank you for refactoring the clipboard util!

**Comment by matejkriz** [comment]
@komret passphrase e2e test (touching receive address flow) failing, please double check 
https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/4415494964

**Comment by bosomt** [comment]
**QA NOK**

when displaying long xpub on device you can only see first characters of xpub, not sure if its expected.
Problematic xpubs: Cardano ADA and BTC taproot

image

**Comment by Hannsek** [comment]
@trezor/qa can you retest this with latest fw prom master? [This](https://github.com/trezor/trezor-firmware/pull/3177) should be the fix.

**Comment by bosomt** [comment]
**QA OK**

checked BTC Taproot xpub and Cardano xpubs

**Info:**
 - Suite version: desktop 23.10.0 (fa5f737baa882c2ee12c368da444743e633d040e)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/23.10.0 Chrome/112.0.5615.204 Electron/24.5.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2B1 2.6.2 regular (revision 6e5967af25de87bebd331d697e6ef940e64286ef)
 - Transport: BridgeTransport 2.0.32
